stories/Button.tsx

```diff
@@ -1,66 +1,124 @@
-import React from "react";
-import PropTypes from "prop-types";
-import "./button.css";
-import { CSSProperties } from "styled-components";
+import styled, { keyframes, css, CSSProperties } from "styled-components";
 
-interface Props {
-  primary: boolean;
-  backgroundColor: string;
-  size: string;
-  label: string;
-  style: CSSProperties | undefined;
-}
-/**
- * Primary UI component for user interaction
- */
-export const Button = ({
-  primary,
-  backgroundColor,
-  size,
-  label,
-  style,
-  ...props
-}: Props) => {
-  const mode = primary
-    ? "storybook-button--primary"
-    : "storybook-button--secondary";
-  return (
-    <button
-      type="button"
-      className={["storybook-button", `storybook-button--${size}`, mode].join(
-        " "
-      )}
-      {...(backgroundColor && { backgroundColor })}
-      {...props}
-    >
-      {label}
-    </button>
-  );
-};
-
-Button.propTypes = {
+interface ButtonProps {
   /**
    * Is this the principal call to action on the page?
    */
-  primary: PropTypes.bool,
+  primary?: boolean;
   /**
    * What background color to use
    */
-  backgroundColor: PropTypes.string,
+  backgroundColor?: string;
   /**
    * How large should the button be?
    */
-  size: PropTypes.oneOf(["small", "medium", "large"]),
+  size?: "small" | "medium" | "large";
   /**
    * Button contents
    */
-  label: PropTypes.string.isRequired,
+  label?: string;
   /**
    * Optional click handler
    */
-  onClick: PropTypes.func,
+  onClick?: () => void;
+  /**
+   * Styling object
+   */
+  style?: CSSProperties;
+}
+
+const getVariantStyles = ({ primary = false }: ButtonProps) =>
+  primary
+    ? css`
+        color: white;
+        background-color: #1ea7fd;
+      `
+    : css`
+        color: #333;
+        background-color: transparent;
+        box-shadow: rgba(0, 0, 0, 0.15) 0px 0px 0px 1px inset;
+      `;
+
+const getSizeStyles = ({ size = "medium" }: ButtonProps) => {
+  switch (size) {
+    case "small": {
+      return css`
+        font-size: 12px;
+        padding: 10px 16px;
+      `;
+    }
+    case "large": {
+      return css`
+        font-size: 16px;
+        padding: 12px 24px;
+      `;
+    }
+    default: {
+      return css`
+        font-size: 14px;
+        padding: 11px 20px;
+      `;
+    }
+  }
 };
 
+/**
+ * Primary UI component for user interaction
+ */
+
+const shake = keyframes`
```

Comment:
YYYYYYAAAAAAAAAAYYYYY The shake is fully back and functional, great use in animation 🐰